src/pages/GovernContent.jsx

```diff
@@ -169,28 +198,35 @@ export default function Govern() {
     }
   }, [])
 
+  useEffect(async () => {
+    let dict = await getCommDict()
+    setCommitDict(dict)
+  }, [])
+
   const owner_address = getWallet().address;
 
   let adjusted = loadedCDPs.filter(value => !value.asaID).map((value) => {
     const cdp_address = cdpGen(owner_address, value.id).address;
-    return {
-      balance: value.collateral == "N/A" ? "N/A" : value.collateral / 1000000,
-      committed:
-        commitment == undefined || commitment[cdp_address] == undefined
-          ? "unknown"
-          : commitment[cdp_address].lastCommitment == -1
-          ? 0
-          : commitment[cdp_address].lastCommitment / 1000000,
-          id: value.id,
-    };
+    if (isFirefox()) {
+      return {
+        balance: value.collateral == "N/A" ? "N/A" : value.collateral / 1000000,
+        committed: <a target="_blank" rel="noreferrer" style={{"text-decoration": "none", "color": "#019fff"}} href="https://governance.algorand.foundation/governance-period-5/governors">See external site</a>
+      }
+    } else {
+      return {
+        balance: value.collateral == "N/A" ? "N/A" : value.collateral / 1000000,
+        committed: commitDict[cdp_address] == 0 || !commitDict[cdp_address] ? 0 : `${(commitDict[cdp_address] / 1000000).toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")}`,
```

Comment:
Could you do the same string formatting for balance? And can you round both numbers to 2 decimals? Otherwise we end up with commas in the decimals area.
Besides that looks perfect! Awesome work!